### PR TITLE
feat: migrate reference line config to mantine

### PR DESCRIPTION
**Pull request by magnew**
Closes: #6266 

### Description:

Migrates reference line config to mantine.

### CODE REVIEW

**Review by IrakliJani — APPROVED** [review]
@magnew code looks great! approving with one UI suggestion.

In terms of UI and since we have a lot of space here - IMO will be better to use https://mantine.dev/core/color-input/


CleanShot 2023-07-26 at 10 01 56@2x

cc: @PriPatel

**Comment by magnew** [comment]
> @magnew code looks great! approving with one UI suggestion.
> 
> In terms of UI and since we have a lot of space here - IMO will be better to use https://mantine.dev/core/color-input/
> 
> CleanShot 2023-07-26 at 10 01 56@2x
> cc: @PriPatel

Sure! I actually thought about it but opted for doing something more similar to what we already had. I'm happy to do this though.

**Review by IrakliJani — APPROVED** [review]
tested in preview, looks great. let's wait for e2e before merging ✅

**Pull request merged**
